exchanges/kline/kline.go

```diff
@@ -606,17 +629,41 @@ func (e *ExchangeIntervals) Construct(required Interval) (Interval, error) {
 		return 0, ErrInvalidInterval
 	}
 
-	if e.supported[required] {
+	if _, ok := e.supported[required]; ok {
 		// Directly supported by exchange can return.
 		return required, nil
 	}
 
 	for x := len(e.aligned) - 1; x > -1; x-- {
-		if e.aligned[x] < required && required%e.aligned[x] == 0 {
+		if e.aligned[x].Interval < required && required%e.aligned[x].Interval == 0 {
 			// Indirectly supported by exchange. Can generate required candle
 			// from this lower time frame supported candle.
-			return e.aligned[x], nil
+			return e.aligned[x].Interval, nil
 		}
 	}
 	return 0, ErrCannotConstructInterval
 }
+
+// GetIntervalResultLimit returns the maximum amount of candles that can be
+// returned for a specific interval. If the individual interval limit is not set,
+// it will be ignored and the global result limit will be returned.
+func (e *ExchangeCapabilitiesEnabled) GetIntervalResultLimit(interval Interval) (int64, error) {
+	if e == nil {
+		return 0, errExchangeCapabilitiesEnabledIsNil
+	}
+
+	val, ok := e.Intervals.supported[interval]
+	if !ok {
+		return 0, errIntervalNotSupported
+	}
+
+	if val > 0 {
+		return val, nil
+	}
+
+	if e.GlobalResultLimit == 0 {
+		return 0, errCannotFetchIntervalLimit
```

Comment:
Is it that it cannot be fetched or is it because the global result limit is unset? You might need such a distinction. Is a fmt.Errorf required here too?